src/components/header/Header.jsx

```diff
@@ -11,13 +14,17 @@ const Header = () => {
 
   return (
     <header className="bg-black header-grid flex p-4">
-      <div className="ml-auto content-end">
-        <ButtonPill text="Log in" onClick={logInOnClick} />
-      </div>
+      {!rootStore.userStore.isLoggedIn ? (
+        <div className="ml-auto content-end">
+          <ButtonPill text="Log in" onClick={logInOnClick} />
+        </div>
+      ) : (
+        <p className="ml-auto content-end text-white">I'm logged in!!!!</p>
+      )}
     </header>
   );
 };
 
-export default Header;
+export default observer(Header);
```

Comment:
If you recall, React components won't update unless you use the `useState` hook. So if we have that constraint, how do we make sure that the header doesn't show the login button when we have the `isLoggedIn` variable in our user account store??

The answer is the `observer` higher order component you see here! You don't need to know all the details of higher order components, but if you wanna read up more about it feel free to do so [here](https://legacy.reactjs.org/docs/higher-order-components.html)